src/IonEventStream.ts

```diff
@@ -240,23 +252,47 @@ export class IonEventStream {
                         eventResult = ionTimestampActual.compareTo(ionTimestampExpected) == 0 ? new ComparisonResult(ComparisonResultType.EQUAL)
                             : new ComparisonResult(ComparisonResultType.NOT_EQUAL, ionTimestampActual + " vs. " + ionTimestampExpected);
                     } else {
-                         eventResult = actualContainerEvent.compare(expectedContainerEvent);
+                            eventResult = actualContainerEvent.compare(expectedContainerEvent);
                     }
 
                     if ((comparisonType == ComparisonType.EQUIVS || comparisonType == ComparisonType.EQUIV_TIMELINE)
                         && eventResult.result == ComparisonResultType.NOT_EQUAL) {
-                        comparisonReport.writeComparisonReport(ComparisonResultType.NOT_EQUAL, eventResult.message, i + 1, j + 1);
+                        if(comparisonReport)
+                            comparisonReport.writeComparisonReport(ComparisonResultType.NOT_EQUAL, eventResult.message, i + 1, j + 1);
                         return new ComparisonResult(ComparisonResultType.NOT_EQUAL);
                     } else if (comparisonType == ComparisonType.NON_EQUIVS && eventResult.result == ComparisonResultType.EQUAL) {
-                        comparisonReport.writeComparisonReport(ComparisonResultType.EQUAL, "Both values are equal in non-equivs comparison.", i + 1, j + 1);
+                        if(comparisonReport)
+                            comparisonReport.writeComparisonReport(ComparisonResultType.EQUAL,
+                                "Both values are equal in non-equivs comparison.", i + 1, j + 1);
                         return new ComparisonResult(ComparisonResultType.EQUAL);
                     }
                 }
             }
             actualIndex++;
             expectedIndex++;
         }
-        return new ComparisonResult(comparisonType == ComparisonType.EQUIVS ? ComparisonResultType.EQUAL : ComparisonResultType.NOT_EQUAL);
+        return new ComparisonResult(comparisonType == ComparisonType.NON_EQUIVS ? ComparisonResultType.NOT_EQUAL : ComparisonResultType.EQUAL);
+    }
+
+    private parseContainer(event: IonEvent): IonEvent[] {
+        let container: IonEvent[] = [];
+        for (let j = 0; j < event.ionValue.length - 1; j++) {
+            container.push(event.ionValue[j]);
+            if (event.ionValue[j].eventType === IonEventType.CONTAINER_START) {
+                j += event.ionValue[j].ionValue.length
```

Comment:
Could you add a semicolon to this line for consistency? (We really should add an automated style checker on this repo.)

Reply:
Yes. As you mentioned earlier should we do it in a separate PR?

Reply:
> As you mentioned earlier should we do it in a separate PR?

I suggest making a new PR any time you find yourself adding substantial new functionality. For example, I think this PR could have been broken up into one PR for the `yargs` framework and another PR for the `compare`/`equivs` implementation.

Incorporating feedback (style changes, refactoring, etc) can all be done as new commits in the same PR.